### DISCUSSION TITLE
Reminder API에서 petPlentId를 쿼리 파라미터로 받을 것인가, 아니면 PathVariable로 받을 것인가?

### DISCUSSION THREAD

**Discussion by rawfishthelgh** [woowacourse-teams/2023-pium#177] [뜨거운 토론]
현재 리마인더의 post, patch url api는 다음과 같습니다

`/reminders/{petPlantId}`

현재 상태에서 요청이 들어오면 `reminders/1` 과 같은 방식으로 들어오게 될 텐데,
이 경우 요청을 보내는 입장에서는 "1번 리마인더를 요청한다" 라고 오해하여 해석할 여지가 생기게 됩니다.

url을 통해 동작을 설명해야 하는 rest api 원칙에 어긋난다는 생각이 들었습니다.
따라서 `/reminders?petPlantId={petPlantId}` 와 같은 쿼리 파라미터를 활용하는 방식으로 url을 변경하는 것은 어떨까 생각이 들었습니다.

**Comment by WaiNaat** [comment]
url만 보고 알 수 없다는 점에서 공감되네요
프론트에서는 지금 `reminders/1`과 같은 url을 아예 사용하고 있지 않아서 변경에 큰 문제는 없을 것 같습니다

**Comment by Choi-JJunho** [comment]
/pet-plants라는 url과 계층을 가지는 형태로 구성하는 방식과 하마드가 말씀하신 쿼리파라미터를 활용하는 방식 두가지가 떠오르네요

## url path 추가
계층형태를 적용한다면 `/pet-plants/{petPlantId}/reminder/{reminderId}` 와 같은 형태로 작성할 수 있을 것 같습니다.

## 쿼라파라미터 적용
쿼리 파라미터로 petPlantId를 관리한다면 하마드가 말씀하신대로
`/reminders?petPlantId={petPlantId}` 와 같은 형태가 될 것 같네요


위 두가지 방식을 고려해봤을 때 저는 하마드의 의견대로 쿼리 파라미터로 명시적으로 적용해주는 편이 더 좋다고 생각됩니다.
![image](https://github.com/woowacourse-teams/2023-pium/assets/49794401/facb7852-5f19-4781-bbc3-2cadefa73fab)

현재 저희 서비스 구성 상 `반려식물의 리마인더` 라는 느낌보다는 `리마인더라는 고유한 자원이 반려식물을 참조한다`라는 생각이 듭니다.
때문에 `/reminders?petPlantId={petPlantId}`를 선호합니다.

**Comment by Choi-JJunho** [comment]
쿼리 파라미터를 이용하는 방식으로 개발 완료되어 close 하겠습니다~
